src/test/thinblock_tests.cpp

```diff
@@ -116,6 +126,231 @@ BOOST_AUTO_TEST_CASE(thinblock_test) {
     CXThinBlock xthinblock7(block, &filter1);
     BOOST_CHECK(xthinblock7.collision);
 
+    /* Message Consistency checking for p2p thinblock messages.  If the messages are not formatted correctly then a
+       a ban will result.
+     */
+    fImporting = false;
+    fReindex = false;
+
+    /** XTHINBLOCK message consistency checks */
+
+    // testing empty missingtx vector
+    CAddress addr1(ipaddress(0xa0b0c001));
+    CNode dummyNode1(INVALID_SOCKET, addr1, "", true);
+    dummyNode1.nVersion = 1;
+
+    CNode::ClearBanned();
+    CDataStream vRecv1(SER_NETWORK, PROTOCOL_VERSION);
+    CXThinBlock xthin = xthinblock;
+    xthin.vMissingTx.clear(); // empty the missingtx vector. This should cause an error.
```

Comment:
very minor formatting nit: this change set mixed trailing comments with comments on separated lines. As stated already I prefer the latter, but at least we should have a consistent approach.